Tasks/NuGetInstaller/NuGetInstaller.ps1

```diff
@@ -62,15 +65,40 @@ if(!$nuGetPath)
     $nuGetPath = Get-ToolPath -Name 'NuGet.exe';
 }
 
+if (-not $nugetPath)
+{
+    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+}
+
 if($nuGetRestoreArgs)
 {
+    if($nuGetRestoreArgs.ToLowerInvariant().Contains("-configfile"))
+    {
+        Write-Warning "ConfigFile was passed as a command line parameter, which may be ignored in certain parts of this task. Please specify the config file in the build definition instead."
+    }
+
     $args = ($args + " " + $nuGetRestoreArgs);
 }
 
-
-if (-not $nugetPath)
+if($nugetConfigPath -ne $env:Build_SourcesDirectory)
 {
```

Comment:
What about for definitions saved pre-91? I think this value might be null/empty. Might be better to:
if ($nuGetConfigPath -and ($nuGetConfigPath -ne $env:Build_SourcesDirectory))